modules/compute/vm-instance/main.tf

```diff
@@ -38,9 +38,15 @@ locals {
 
   # both of these must be false if either compact placement or preemptible/spot instances are used
   # automatic restart is tolerant of GPUs while on host maintenance is not
-  automatic_restart           = local.compact_placement || var.spot ? false : null
+  automatic_restart_default   = local.compact_placement || var.spot ? false : null
```

Comment:
There is an existing problem with `local.compact_placement` that we should address:

```
  compact_placement = try(var.placement_policy.collocation, null) != null
```

This assumes that any value at all for `var.placement_policy.collocation` implies that the user wants compact placement. There are 2 alternatives:

1. They have asked for a "spread" placement policy (the opposite of compact)
2. They have asked for an invalid placement policy.

This could probably with some validation blocks on `var.placement_policy` (something like `var.placement_policy == null || contains(var.placement_policy.collocation, ["valid", "values"]`.

I will also add this PR to a tracker for reasons we might consider upgrading to 1.3 (not obvious it helps, but it might).

Reply:
The documentation is suggestive (though not definitive) that a spread placement policy should have not require specifying a restart policy. The [section for applying a compact policy][compact] to an existing instance includes the text:

> For compact placement policies, you must include the --maintenance-policy=TERMINATE and --no-restart-on-failure flags.

The [section for spread policies][spread] does not include this text.

[compact]: https://cloud.google.com/compute/docs/instances/define-instance-placement#apply-new-instance
[spread]: https://cloud.google.com/compute/docs/instances/define-instance-placement#apply-existing-instance

Reply:
Discussed offline, the issue is not blocking for this PR. Tracked in separate issue.